src/components/RadioButtonGroup/index.js

```diff
@@ -45,8 +61,62 @@ class RadioButtonGroup extends Component {
         return undefined;
     }
 
+    getCheckedOptionRef() {
+        const { value, options } = this.props;
+        const currentOptionIndex = options.findIndex(option => isOptionChecked(option, value));
+        return currentOptionIndex !== -1 ? this.optionsRefs[currentOptionIndex] : null;
+    }
+
+    generateRefsForOptions() {
+        const { options } = this.props;
+        return options.reduce(list => [...list, React.createRef()], []);
+    }
+
+    addRefsToOptions(options) {
+        return options.reduce(
+            (list, option, index) => [
+                ...list,
+                {
+                    optionRef: this.optionsRefs[index],
+                    ...option,
+                },
+            ],
+            [],
+        );
+    }
+
+    isMarkerActive() {
+        const { value, options } = this.props;
+        const reducer = (accumulator, option) => accumulator || isOptionChecked(option, value);
+
+        return options.reduce(reducer, false);
+    }
+
+    handleChange(event) {
+        const { onChange } = this.props;
+        onChange(event);
+        setTimeout(() => {
+            this.updateMarker();
+        }, 0);
+    }
+
+    updateMarker() {
+        const activeOptionRef = this.getCheckedOptionRef();
+        if (activeOptionRef !== null) {
+            this.setState({
+                markerLeft: activeOptionRef.current.offsetLeft,
+                markerWidth: Math.max(
+                    activeOptionRef.current.offsetWidth,
+                    activeOptionRef.current.clientWidth,
+                ),
+            });
+        }
+    }
+
     render() {
-        const { style, label, required, error, onChange, options, value, id } = this.props;
+        const { style, label, required, options, error, value, id } = this.props;
+        const optionsWithRefs = this.addRefsToOptions(options);
```

Comment:
this function make an iteration and is called every time the component rerender, so this need to be computed only one time and when the options are changed dynamically(check and recompute in componentDidUpdate when prev options are different from current options).